stellargraph/data/explorer.py

```diff
@@ -451,24 +418,30 @@ def transition_probability(nn):
                 current_node = node
 
                 for _ in range(length - 1):
-                    neighbours = self.graph.neighbors(
-                        current_node, use_ilocs=True, include_edge_weight=weighted
-                    )
-
-                    if not neighbours:
-                        break
-
                     # select one of the neighbours using the
                     # appropriate transition probabilities
-                    choice = naive_weighted_choices(
-                        rs, (transition_probability(nn) for nn in neighbours),
-                    )
+                    if weighted:
+                        neighbours, weights = self.graph.neighbor_arrays(
+                            current_node, include_edge_weight=True, use_ilocs=True
+                        )
+                    else:
+                        neighbours = self.graph.neighbor_arrays(
+                            current_node, use_ilocs=True
+                        )
+                        weights = np.ones(neighbours.shape, dtype=weight_dtype)
```

Comment:
Does the 10x speedup still apply to this case?

Reply:
yep, in this PR `weighted=None` and `weighted=True` take the same amount of time. For `weighted=None` the `weights` array is just created here instead of in `neighbor_arrays`. 

Buuut, actually in develop `weighted=True` takes alot longer so its more like a 100x speedup.

Also casting the `iq` and `ip` instead of `weights` has made the `weighted=None` case more like a 20x speedup, and that's without your suggestion!

Reply:
Nice!

Could you update the PR title (and thus merge commit message) and description as appropriate to reflect the new state?